### PR TITLE
Foldmason

### PR DESCRIPTION
**Pull request by luisas**
## PR checklist

Add foldmason easy-msa module.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/foldmason/easymsa/main.nf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        foldmason: \$(foldmason | grep "foldmason Version:" | cut -d":" -f 2 | awk '{\$1=\$1;print}')
+        pigz: \$(echo \$(pigz --version 2>&1) | sed 's/^.*pigz\\w*//' ))
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.fa${compress ? '.gz' : ''}
```
Comment (SPPearce) [inline modules/nf-core/foldmason/easymsa/main.nf:46]: nf-test doesn't like gz files that aren't actually made by gzip, it gives a weird error (printing the full file paths in the snapshot). This is our workaround:
```suggestion
    echo "" | gzip > ${prefix}.fa${compress ? '.gz' : ''}
```

File: modules/nf-core/foldmason/easymsa/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    }
+            }
+        }
+
+        when {
+            params {
+            }
+            process {
+                """
+                    input[0] = UNTAR.out.untar.map { meta,dir -> [[ id:'test' ], file(dir).listFiles().collect()]}
```
Comment (SPPearce) [inline modules/nf-core/foldmason/easymsa/tests/main.nf.test:32]: I feel like this should be the same?
```suggestion
                    input[0] = UNTAR.out.untar.map { meta,dir -> [meta, file(dir).listFiles().collect()]}
```

File: modules/nf-core/foldmason/easymsa/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success  },
+                { assert snapshot(process.out.msa3di).match("alignment_uncompressed_3di")},
+                { assert snapshot(process.out.msaAA).match("alignment_uncompressed_AA")},
+                { assert snapshot(process.out.versions).match("versions0") }
```
Comment (SPPearce) [inline modules/nf-core/foldmason/easymsa/tests/main.nf.test:43]: Might need aligning properly, but can you put them into one snapshot:
```suggestion
                { assert snapshot(
                  process.out.msa3di,
                  process.out.msaAA,
                  process.out.versions).match() }
```

**Comment by SPPearce** [comment]
Looks generally fine, just we only use lower_snake_case for emit channels, would `msa_aa` suffice?

**Comment by luisas** [comment]
@SPPearce thanks a lot for the comments :) I updated what you suggested and I could use a review when you have some spare time!

**Comment by luisas** [comment]
It would :) I updated it now

**Review by SPPearce — DISMISSED** [review]

File: modules/nf-core/foldmason/easymsa/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert process.success  },
+                { assert snapshot(
+                  process.out.msa_3di,
+                  process.out.msa_aa,
+                  process.out.versions).match() }
+                )
```
Comment (SPPearce) [inline modules/nf-core/foldmason/easymsa/tests/main.nf.test:45]:
```suggestion
                { assert snapshot(process.out).match() }
                )
```

File: modules/nf-core/foldmason/easymsa/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success  },
+                { assert snapshot(process.out.msa_3di).match("alignment_compressed_3di")},
+                { assert snapshot(process.out.msa_aa).match("alignment_compressed_AA")},
+                { assert snapshot(process.out.versions).match("versions1") }
```
Comment (SPPearce) [inline modules/nf-core/foldmason/easymsa/tests/main.nf.test:79]:
```suggestion
                { assert snapshot(process.out).match() }
```

File: modules/nf-core/foldmason/easymsa/tests/main.nf.test.snap
Comment (SPPearce) [inline modules/nf-core/foldmason/easymsa/tests/main.nf.test.snap:1]: This doesn't look like it has been regenerated.

**Review by SPPearce — CHANGES_REQUESTED** [review]
Sorry, pressed approve too fast.
Can you please add a stub test too.

File: modules/nf-core/foldmason/easymsa/environment.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/environment-schema.json
+name: "foldmason_easymsa"
+channels:
+  - conda-forge
+  - bioconda
+  - defaults
+dependencies:
+  - "bioconda::foldmason=1.763a428"
```
[the remark this answers is not in the document]
Comment (luisas) [reply modules/nf-core/foldmason/easymsa/environment.yml:9]: It was a mistake! Thanks for spotting it

File: modules/nf-core/foldmason/easymsa/tests/main.nf.test.snap
Comment (luisas) [reply modules/nf-core/foldmason/easymsa/tests/main.nf.test.snap:1]: Now this is fixed too

**Review by SPPearce — APPROVED** [review]
Thank you!

**Pull request merged**